marker_api/src/interface.rs

```diff
@@ -78,6 +79,12 @@ macro_rules! export_lint_pass {
                 extern "C" fn info() -> $crate::LintPassInfo {
                     super::__MARKER_STATE.with(|state| state.borrow_mut().info())
                 }
+                extern "C" fn check_crate<'ast>(
```

Comment:
I wonder if this PR also changes the behaviour of "check_item"? I think that method must also traverse the root crate module. In general, I think `check_crate` will be very rarely needed, except for lints that specially handle the crate's root module. Anyhow, the crate's root module should not be anything special in the context of `check_item`

Reply:
The root module is passed to `check_item` like any other module. The goal was to handle the root module, like a normal module, to make linting easy.